core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -145,17 +149,22 @@ private LocalDate getCurrentDayDate(UserAnonymizedDTO userAnonymized)
 
 	private <T extends IntervalActivity> void addMissingInactivity(Goal activeGoal, ZonedDateTime dateAtStartOfInterval,
 			Set<T> activityEntitiesAtDate, UserAnonymizedDTO userAnonymized,
-			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier)
+			BiFunction<Goal, ZonedDateTime, T> inactivityEntitySupplier, Consumer<T> existingEntityInactivityCompletor)
 	{
-		if (!containsActivityForGoal(activityEntitiesAtDate, activeGoal))
+		Optional<T> activityForGoal = getActivityForGoal(activityEntitiesAtDate, activeGoal);
+		if (activityForGoal.isPresent())
+		{
+			existingEntityInactivityCompletor.accept(activityForGoal.get());
```

Comment:
In case of week activities, it may be that there was activity recorded at some days in the week, Wednesday and Thursday for example. When retrieving the week, inactivity days should be added, in this case Sunday, Monday, Tuesday, Friday, Saturday.